Patch-Source: https://github.com/SCST-project/scst/pull/285/commits/ad00df2630eb54775a96337e9e1f89676cdc9e73
Backported for 2.2.1.
---
From ad00df2630eb54775a96337e9e1f89676cdc9e73 Mon Sep 17 00:00:00 2001
From: Gleb Chesnokov <gleb.chesnokov@scst.dev>
Date: Fri, 18 Apr 2025 15:51:09 +0300
Subject: [PATCH] iscsi-scstd: Fix redefinition of bool when building with GCC
 15

GCC 15 (C23) now reserves bool/true/false, so the local typedef in
iscsid.d breaks the build:

    error: cannot use keyword 'false' as enumeration constant
    error: expected ';', identifier or '(' before 'bool'

Include <stdbool.h> and drop the typedef.
---
 iscsi-scst/usr/iscsid.h | 5 +----
 1 file changed, 1 insertion(+), 4 deletions(-)

diff --git a/usr/iscsid.h b/usr/iscsid.h
--- a/usr/iscsid.h
+++ b/usr/iscsid.h
@@ -23,6 +23,7 @@
 #include <sys/poll.h>
 #include <assert.h>
 #include <netdb.h>
+#include <stdbool.h>
 
 #include "types.h"
 #ifdef INSIDE_KERNEL_TREE
@@ -34,10 +35,6 @@
 #include "param.h"
 #include "misc.h"
 
-#ifndef bool
-typedef enum {false = 0, true} bool;
-#endif
-
 #define sBUG() assert(0)
 #define sBUG_ON(p) assert(!(p))
 
